### PR TITLE
Add "map state" map data count

### PR DESCRIPTION
**Pull request by soyarsauce**
### What this PR does

Fixes https://github.com/TerriaJS/terriajs/issues/4450

* Add "map state" map data count to highlight state of map data

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ Change Log
 * Use `1.x.x` of `karma-sauce-launcher` to fix CI build failures
 * Stop unknown icons specified in config.json from crashing UI
 * Creates a `ShadowTraits` class that is shared by `GltfCatalogItem` and `Cesium3DTilesCatalogItem`.
+* Add "Upload" to tour points
```
Comment (na9da) [inline CHANGES.md]: Stray entry?

File: lib/ReactViews/BottomDock/MapDataCount.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    // Can't simply use number of items given they can exist in workbench
+    // without being shown on map
+    const numberOfDatasets = terria.workbench.items.filter(item => item.show)
+      .length;
+    const hasMapData = numberOfDatasets !== 0;
+    const dataset = t("countDatasets.datasetSingular");
+    const datasetPlural = t("countDatasets.datasetPlural");
+    const datasetString =
+      numberOfDatasets > 1 || numberOfDatasets === 0 ? datasetPlural : dataset;
+    const mapDataText = hasMapData
```
Comment (na9da) [inline lib/ReactViews/BottomDock/MapDataCount.jsx:28]: Would it be more appropriate to rewrite this in the i18next interpolation style?

File: lib/Styled/ButtonAsLabel.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// import React from "react";
```
Comment (na9da) [inline lib/Styled/ButtonAsLabel.jsx]: Remove the commented imports?

**Comment by na9da** [comment]
Hey, looks great overall, just a few minor suggestions.

File: lib/ReactViews/BottomDock/MapDataCount.jsx (hunk 2 above)
Comment (soyarsauce) [reply lib/ReactViews/BottomDock/MapDataCount.jsx:28]: Functionally the same but there could be the case where the grammar/order works differently for a different language so I've moved a tiny bit of the logic into the translation file

File: CHANGES.md (hunk 1 above)
Comment (soyarsauce) [reply CHANGES.md]: Removed

File: lib/Styled/ButtonAsLabel.jsx (hunk 3 above)
Comment (soyarsauce) [reply lib/Styled/ButtonAsLabel.jsx]: Removed

**Comment by soyarsauce** [comment]
Thanks @na9da - all addressed, please approve if it looks good

**Comment by soyarsauce** [comment]
(& merge)

**Review by na9da — APPROVED** [review]

**Pull request merged**